keps/sig-cli/3895-kubectl-delete-interactivity/README.md

```diff
@@ -0,0 +1,926 @@
+However, that comes along with a technical limitation that `Visit` function does not support traversing twice
+as stated in [here](https://github.com/kubernetes/kubernetes/blob/6d7a9048b67d81e57923892650a23636f1afba42/staging/src/k8s.io/cli-runtime/pkg/resource/result.go#L91-L94).
+But we also cannot use `Infos` function as suggested in the link above, because that unties the builder configuration
+and deletion logic.
+
+In order to overcome this limitation, this feature will use separate builder to show resources as preview. Current builder
+will continue being used as deletion traverse, other builder will be used for preview. Builders must match in terms of 
+configurations to prevent such discrepancies and required tests will be added to assure that configurations are same or not.
+Moreover, deletion will only be performed to the resources whose `uidMap` match the ones in the preview list.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+As a user, I want to have an option via a flag to list all the resources being deleted as preview and
+proceed to actual deletion only after explicit confirmation.
+
+#### Story 1
+As filed in this issue https://github.com/kubernetes/kubectl/issues/1365;
+user tries to delete all persistentvolumeclaims in a namespace;
+```sh
+$ kubectl delete --all pvc -n yournamespace
+```
+
+However, mistyping `pv` instead of `pvc` deletes all persistentvolumes in the cluster, 
+since persistentvolumes are non-namespaced resources.
+
+#### Story 2
+As filed in this issue https://github.com/kubernetes/kubectl/issues/696;
+user executes `kubectl delete --all namespaces` without enough knowledge about the
+consequences of this command.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+- In terms of backwards compatibility, this flag's default will always be false.
+- If user cancels deletion, exit code will be 0.
+- This flag will not be used with `--raw` flag. 
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+Only standard input via terminal will be supported because main motivation of this flag
+is to provide a way to users. That would be a risk if there is exceptional inputs whose are available
+for users that we can not handle.
```

Comment:
I'd propose stating this something like:

This mechanism will apply only to client-side operation (`kubectl delete`) and as mentioned in non-goals section is not meant to solve server-side protection, which can be addressed using custom [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook), for example.